ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -98,6 +106,11 @@ protected PubsubMessage processElement(PubsubMessage message)
       throw new MessageShouldBeDroppedException();
     }
 
+    if (MessageScrubber.shouldScrubAndWriteToErrors(attributes, json)) {
+      // Write the message to error output.
+      throw new MessageToErrorsException("Message should be scrubbed");
+    }
+
```

Comment:
I think there's enough complexity here now that we should refactor all the responsibility into `MessageScrubber` itself. So in ParsePayload, we would just have a single call:

```
// For various messages signatures, this may redact fields from json or throw an exception
// that causes the message to be dropped or routed to error output.
MessageScrubber.scrub(attributes, json);
```

Then within `scrub`, each conditional would either modify the json payload, throw `MessageShouldBeDroppedException` or a new `AffectedByBugException` where we pass the bug number into the error message. 

To avoid having to repeat the bug number several times (I'm getting worried at this point that there's a high likelihood we'll have a copy/paste problem where we end up incrementing the wrong counter, etc.), perhaps we should have a new method:

```
public enum ScrubAction {
  DROP,
  REDACT,
  SEND_TO_ERRORS
}

public static void handleBug(String bugNumber, ScrubAction action) {
  Metrics.counter(MessageScrubber.class, "bug_" + bugNumber).inc();
  switch(action) {
    case DROP:
      throw new MessageShouldBeDroppedException();
    case SEND_TO_ERRORS:
      throw new AffectedByBugException(bugNumber);
    case REDACT:
      return;
  }
}
```

Reply:
Although now that I write this and consider @badboy's comment, we probably should delay the handling for this bug until near the end of this transform so that we can take advantage of the logic later on for parsing clientId out of various potential locations within the payload.

If some glean clients also are having their id set to "coffee", we should include those too.

I think we want to drop toxic messages as early as possible in case they raise some exeption within the transform that causes them to go to error output rather than being dropped. But for messages that we want to send to error output rather than dropping, we can wait until the end of the method. So I think that's an argument for having a `shouldDrop` method and a `shouldWriteToErrors` method.

Reply:
I ended up refactoring `MessageScrubber` a little and still keep separate methods that can be called for scrubbing or writing messages to error. I liked the idea of throwing Exceptions in `MessageScrubber` since that reduces the complexity a little, but I'm also open to change them back to being `boolean` and throw exceptions when parsing the payload.